pkg/secrets/pushtofile/secrets_groups.go

```diff
@@ -0,0 +1,220 @@
+package pushtofile
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/log/messages"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	CONJUR_SECRETS_PREFIX             = "conjur.org/conjur-secrets."
+	CONJUR_SECRETS_POLICY_PATH_PREFIX = "conjur.org/conjur-secrets-policy-path."
+	SECRET_FILE_PATH_PREFIX           = "conjur.org/secret-file-path."
+	SECRET_FILE_FORMAT_PREFIX         = "conjur.org/secret-file-format."
+	SECRET_FILE_TEMPLATE_PREFIX       = "conjur.org/secret-file-template."
+
+	DEFAULT_FILE_PERMS os.FileMode = 0777
+	DEFAULT_FILE_PATH  string      = "/conjur/secrets/"
+)
+
+type SecretsFileFormat int
+
+const (
+	FILE_FORMAT_NONE SecretsFileFormat = iota
+	FILE_FORMAT_YAML
+	FILE_FORMAT_JSON
+	FILE_FORMAT_DOTENV
+	FILE_FORMAT_BASH
+	FILE_FORMAT_PLAINTEXT
+)
+
+// SecretsPaths comprises Conjur variable paths for all secrets in a secrets
+// group, indexed by secret name.
+type SecretsPaths map[string]string
```

Comment:
So is this a map of the secret alias to the full path? Would it be better to have this as a struct
with the ID and alias 
`type SecretSpec struct {
	Id    string
	Alias string
}`

Reply:
I just followed the spec - and I think it makes more sense for this to be a map.

Reply:
I disagree and think a map is better suited here. 1) Disagree on added cognitive load (for whom and how?) 2) It maps (key = secret name, val = conjur secret path) so `SecretPaths` seems like an appropriate name. I changed it to avoid prolonged pointless bikeshedding